stubs/networkx/@tests/test_cases/check_dispatch_decorator.py

```diff
@@ -0,0 +1,23 @@
+from typing_extensions import assert_type
+
+from networkx.utils.backends import _dispatch
+
+
+@_dispatch
+def some_method(int_p: int, str_p: str) -> float:
+    ...
+
+
+# Wrong param / order
+some_method("", 0)  # type: ignore
+# backend is kw-only
+some_method(0, "", None)  # type: ignore
+# No backend means no backend_kwargs allowed
+some_method(0, "", backend_arg="")  # type: ignore
+some_method(0, "", backend=None, backend_arg="")  # type: ignore
+
+# Correct usage
+assert_type(some_method(0, ""), float)
+# type system doesn't allow this yet (see comment in networkx/utils/backends.pyi)
+# assert_type(some_method(0, "", backend=None), float)
```

Comment:
This would require some way of concatenating `backend` to `_P.kwargs`. Which to my understanding, isn't currently possible and is a rejected idea in https://peps.python.org/pep-0612/#concatenating-keyword-parameters